massa-api/src/public.rs

```diff
@@ -381,11 +381,46 @@ impl MassaRpcServer for API<Public> {
             })
             .collect::<BTreeMap<_, _>>();
 
+        let current_cycle = last_slot
+            .unwrap_or_else(|| Slot::new(0, 0))
+            .get_cycle(api_settings.periods_per_cycle);
+
+        let cycle_duration = match api_settings.t0.checked_mul(api_settings.periods_per_cycle) {
+            Ok(cycle_duration) => cycle_duration,
+            Err(e) => return Err(ApiError::TimeError(e).into()),
+        };
+
+        let previous_cycle = current_cycle.saturating_sub(1);
+
+        let elapsed_time_before_current_cycle = match if previous_cycle == 0 {
+            Ok(MassaTime::from(0))
+        } else {
+            cycle_duration.checked_mul(previous_cycle)
```

Comment:
Maybe I'm misunderstanding but I find the calculation to be false because we have a cycle 0. Let me try to explain with an example : 
```
genesis = 1000
t0 = 100
period_per_cycle = 100
slot = (280, 3)
current_cycle = 2
previous_cycle = 1
cycle_duration = (100 * 100) = 10000
elapsed_time_before_current_cycle = 10000
```

`current_cycle_time` is `10000 + 1000 = 11000` but should be in fact `21000` because in cycle 2 already 2 cycles has elapsed.

Reply:
![image](https://user-images.githubusercontent.com/22281426/215450514-15486ce0-9344-40e8-98c0-0b442ef4b060.png)

I have `genesis_timestamp` but how can I find the duration between It and the start of cycle 0 ?

Reply:
Start of cycle 0 == genesis_timestamp